pkg/plugins/runtime/universal/outbound.go

```diff
@@ -0,0 +1,42 @@
+package universal
+
+import (
+	"context"
+
+	"github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+	"github.com/kumahq/kuma/pkg/dns"
+)
+
+func UpdateOutbounds(ctx context.Context, rm manager.ResourceManager, vips dns.VIPList) error {
+	meshes := &mesh.MeshResourceList{}
+	if err := rm.List(ctx, meshes); err != nil {
+		return err
+	}
+	for _, m := range meshes.Items {
+		dpList := &mesh.DataplaneResourceList{}
+		if err := rm.List(ctx, dpList, store.ListByMesh(m.Meta.GetName())); err != nil {
+			return err
+		}
+		externalServices := &mesh.ExternalServiceResourceList{}
+		if err := rm.List(ctx, externalServices, store.ListByMesh(m.Meta.GetName())); err != nil {
+			return err
+		}
+		dpsUpdated := 0
+		for _, dp := range dpList.Items {
+			if dp.Spec.Networking.GetTransparentProxying() == nil {
+				continue
+			}
+			dp.Spec.Networking.Outbound = dns.VIPOutbounds(dp.Meta.GetName(), dpList.Items, vips, externalServices.Items)
+			if err := rm.Update(ctx, dp); err != nil {
```

Comment:
Please do a check before this line that we execute Update only if outbound changed, otherwise we will be executing excessive updates which will then trigger mesh hash change

Reply:
Good point! Thanks